src/client/opamClient.ml

```diff
@@ -1001,7 +1018,26 @@ let install_t t ?ask atoms add_to_roots ~deps_only =
   let t = {t with available_packages = lazy available_packages} in
 
   if pkg_new = [] && OpamPackage.Set.is_empty pkg_reinstall then t else
-
+  let atoms =
+    if assume_built then
+      let pkg_of_atoms =
+        OpamPackage.Set.filter
+          (fun p -> List.exists (fun a -> OpamFormula.check a p) atoms)
+          t.packages
+      in
+      let universe =
+        OpamSwitchState.universe t
+          ~requested:(OpamPackage.names_of_packages pkg_of_atoms)
+          Install
+      in
+      let fixed =
+        OpamPackage.Set.of_list
+          (OpamSolver.dependencies ~build:true ~post:true
```

Comment:
Shoudln't `post` be `false` here ? We _could_ also not consider `build` deps, since they don't trigger recompilations, but I'd rather have them, seems simpler.